ui/src/hooks/useKubernetes.ts

```diff
@@ -333,6 +333,11 @@ export function handleTerminalMessage(ws: WebSocket, event: MessageEvent, create
           terminalStore.setTerminalConfig('backspaceAsCtrlH', value);
         }
 
+        if (sessionInfo.ctrlCAsCtrlZ) {
+          const value = sessionInfo.ctrlCAsCtrlZ ? '1' : '0';
+          terminalStore.setTerminalConfig('ctrlCAsCtrlZ', value);
+        }
+
         if (setting.value.SECURITY_SESSION_SHARE && share) {
           operatedNode.enableShare = true;
         }
```

Comment:
There's no apparent issue to report here. However, you should make sure that all variables and configurations are properly defined before accessing them. It would also be beneficial to add error handling where necessary. Additionally, consider using the `try...catch` block in cases when unexpected events occur.